extensions/default/src/DicomWebDataSource/wado/retrieveMetadataLoader.js

```diff
@@ -12,20 +12,27 @@ export default class RetrieveMetadataLoader {
    * @param {Array} studyInstanceUID Study instance ui to be retrieved
    * @param {Object} [filters] - Object containing filters to be applied on retrieve metadata process
    * @param {string} [filter.seriesInstanceUID] - series instance uid to filter results against
-   * @param {Function} [sortSeries] - Custom sort function for series
+   * @param {Object} [sortCriteria] - Custom sort criteria used for series
+   * @param {Function} [sortFunction] - Custom sort function for series
+   * @param {number} [largeStudySeriesCountThreshold ] - Threshold used for classify a study as a large studies
+   * @param {number} [initialSeriesFetchSize] - In case of a large study, fetch first <firstGroupSize> series to display the first image fast to the user
    */
   constructor(
     client,
     studyInstanceUID,
     filters = {},
-    sortCriteria,
-    sortFunction
+    sortCriteria = undefined,
+    sortFunction = undefined,
+    largeStudySeriesCountThreshold = 100,
```

Comment:
This is probably a bit high - I would guess around 50 is where it starts hitting issues.  Also, the value should come from the datasource being queried, and not be provided globally - dififerent data sources might behave sufficiently different that it doesn't make sense to apply the same behaviour to them.